zuul-core/src/main/java/com/netflix/zuul/netty/server/ClientRequestReceiver.java

```diff
@@ -267,6 +269,10 @@ private HttpRequestMessage buildZuulHttpRequest(
 
         // This is the only way I found to get the port of the request with netty...
         final int port = channel.attr(SourceAddressChannelHandler.ATTR_SERVER_LOCAL_PORT).get();
+        final SocketAddress remoteAddress = channel.attr(SourceAddressChannelHandler.ATTR_REMOTE_ADDR).get();
+        if (remoteAddress instanceof InetSocketAddress) {
+            context.set(CommonContextKeys.REMOTE_PORT, ((InetSocketAddress) remoteAddress).getPort());
```

Comment:
I believe what you're trying to do exists in the form of `HttpRequestMessageImpl#clientRemoteAddress`. 
The issue is likely that I set that value to `ATTR_LOCAL_ADDR` instead of `ATTR_REMOTE_ADDR`, as you correctly do here.
See https://github.com/Netflix/zuul/blob/d8838c5131bf1ea58cd141bd2a8cd0ff8ea1ab95/zuul-core/src/main/java/com/netflix/zuul/netty/server/ClientRequestReceiver.java#L271
I'd probably try updating that and update the tests, which might fix the issue.